File: bindings/nostr-js/examples/nip44.js
const { Keys, PublicKey, nip44_encrypt, nip44_decrypt, NIP44Version, loadWasmAsync } = require("../");

async function main() {
    await loadWasmAsync();

    let keys = Keys.generate();
    
    let public_key = PublicKey.fromHex("79be667ef9dcbbac55a06295ce870b07029bfcdb2dce28d959f2815b16f81798");

    let ciphertext = nip44_encrypt(keys.secretKey, public_key, "my message", NIP44Version.V2)
    console.log("Encrypted: " + ciphertext)

    let plaintext = nip44_decrypt(keys.secretKey, public_key, ciphertext)
    console.log("Decrypted: " + plaintext)
}

main();

File: book/snippets/swift/NostrSnippets/Sources/Nip44.swift
import Foundation
import NostrSDK

func nip44() throws {
    let keys = Keys.generate()

    let publicKey = try PublicKey.parse(publicKey: "79be667ef9dcbbac55a06295ce870b07029bfcdb2dce28d959f2815b16f81798")

    let ciphertext = try nip44Encrypt(secretKey: keys.secretKey(), publicKey: publicKey, content: "my message", version: Nip44Version.v2)
    print("Encrypted: \(ciphertext)");

    let plaintext = try nip44Decrypt(secretKey: keys.secretKey(), publicKey: publicKey, payload: ciphertext)
    print("Decrypted: \(plaintext)");
}
